src/core/ext/client_config/client_channel.c

```diff
@@ -772,13 +817,69 @@ static void cc_start_transport_stream_op(grpc_exec_ctx *exec_ctx,
   GPR_TIMER_END("cc_start_transport_stream_op", 0);
 }
 
+// Gets data from the service config.  Invoked when the resolver returns
+// its initial result.
+static void read_service_config(grpc_exec_ctx *exec_ctx, void *arg,
+                                grpc_error *error) {
+  grpc_call_element *elem = arg;
+  channel_data *chand = elem->channel_data;
+  call_data *calld = elem->call_data;
+  // If this is an error, there's no point in looking at the service config.
+  if (error != GRPC_ERROR_NONE) return;
+  // Get the method config table from channel data.
+  gpr_mu_lock(&chand->mu);
```

Comment:
As above, I'm not sure why these locking comments are needed.  Is there something particularly confusing about this code that's not obvious to me?  (Please do push back if there is.)